### PR TITLE
Status bar redesigned (closes #2510)

### LINKED ISSUE #2510: Redesign status bar to enable debug panel on small screens and emulated mobile devices
### Are you requesting a feature or reporting a bug?
A bug

### What is the current behavior?
The "debug" panel does not fit and thus the "Unlock page" button is invisible

### What is the expected behavior?
The "Unlock page" button is invisible

### How would you reproduce the current behavior (if this is a bug)?
Run the below test

#### Provide the test code and the tested page URL (if applicable)

Tested page URL: http://mdn.github.io/simple-shared-worker/index2.html

Test code

```js
import { Selector } from 'testcafe';

fixture `Shared worker test`
.page `http://mdn.github.io/simple-shared-worker/index2.html`;

test.only('Test shared worker', async (t) => {
	const number = '4';
	const input = Selector('input[id*=number3]');
	await t
		.debug()
		.typeText(input, number)
});

```

### Specify your

* testcafe version: 0.20

### PR DESCRIPTION
**Pull request by Dmitry-Ostashev**
Use the following test to check this pull request:

```
import { Selector } from 'testcafe';

fixture`Fixture 1`
    .page`https://interactive-examples.mdn.mozilla.net/pages/css/z-index.html`;

test('Test 1', async t => {
    await t
        .eval(() => {
            const statusBar = window['%testCafeDriverInstance%'].statusBar;
            statusBar.setStatusPrefix('TestCafe Script execution');
        });

    await t
        .click(Selector('code').withText('z-index: 5;'))
        .click('.position2')
        .debug()
        .expect(Selector('#input').exists).ok({timeout: 5000});
});
```
This test allows to check the Debug state and Waiting state (failed and successful).
Run this test on a desktop and mobile device. Try to resize the browser window.

### CODE REVIEW

**Comment by Dmitry-Ostashev** [comment]
image

image

**Comment by AndreyBelym** [comment]
Test build:

https://github.com/AndreyBelym/testcafe/releases/download/status_bar-redisign-1/testcafe-1.6.1.tgz
https://git.io/JeoL5

**Comment by Dmitry-Ostashev** [comment]
@testcafe-build-bot  retest

**Comment by Dmitry-Ostashev** [comment]
@testcafe-build-bot retest

**Comment by AndreyBelym** [comment]
The new test doesn't work:

https://dev.azure.com/DevExpressTestCafe/testcafebuildbot-testcafe/_build/results?buildId=4113&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=35f53584-79fc-539f-463c-6471d0663a3a&l=1143
https://dev.azure.com/DevExpressTestCafe/testcafebuildbot-testcafe/_build/results?buildId=4114&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=35f53584-79fc-539f-463c-6471d0663a3a&l=1144

**Comment by Dmitry-Ostashev** [comment]
@testcafe-build-bot retest

**Comment by Dmitry-Ostashev** [comment]
@testcafe-build-bot retest

**Comment by Dmitry-Ostashev** [comment]
@testcafe-build-bot retest

**Comment by helen-dikareva** [comment]
1) Status Bar is always green after successful element waiting 
```js
test('t', async t => {
    await t
        .click(Selector('code').withText('z-index: 5;'))
        .debug();
});
```
image

And I don't like 3px bottom margin for colored part of status bar in this case.

2) Some extra margings in IE
image
image

3) we don't add prefix for loading state, but we should do it for e.g. `navigateTo` action in `RunTestCafe script`
image

4) the following test set Status Bar in "green state" even before the wait is completed
```js
test('t', async t => {
    await t
        .click(Selector('code').withText('z-index: 5;'))
        .click('.position2')
        .debug();
});
```
It likely doesn't related with PR and if so we should create a separate issue with this reproducible example.
image

**Comment by helen-dikareva** [comment]
I like the way The new status bar looks

**Comment by Dmitry-Ostashev** [comment]
> I like the way the new status bar looks on mobile devices ❤️
> But there is a little problem: when I run a test from the desciption in emulation or on my phone I see 'waiting for the element' state then 'debug' state, but 'assertion waiting' state isn't shown after I press `resume` button (see [video](http://bit.ly/35qaA9y))

We discussed it with Alexey and we didn't found any perfect solution for such case (e.g. "how to hide/show the status bar on mobile devices to be able to tab an element at the bottom of the page"). So currently it works as similar as on the desktop: tap to the center of the page to show the status bar, tap to the status bar to hide it. 

Any suggestions would be appreciated :-)

**Comment by Dmitry-Ostashev** [comment]
> 1. Status Bar is always green after successful element waiting
> 
> 
> And I don't like 3px bottom margin for colored part of status bar in this case.
> 
>  the following test set Status Bar in "green state" even before the wait is completed
> 
> ```js
> test('t', async t => {
>     await t
>         .click(Selector('code').withText('z-index: 5;'))
>         .click('.position2')
>         .debug();
> });
> ```
> 
> It likely doesn't related with PR and if so we should create a separate issue with this reproducible example.

I've created a separate issue for this case: https://github.com/DevExpress/testcafe/issues/4529

**Comment by helen-dikareva** [comment]
1) Problems with IE fixed. 
2) Problem with unexpected green color extracted in [a separate issue](https://github.com/DevExpress/testcafe/issues/4529)
3) Problem with hidden status bar on mobile device after click on any its buttons - it's a separate problem too. I think we should not hide status bar after click on its button until you move out mouse cursor. Hide/show functionality will be also quite useful.
4) Prefix for loading state still doesn't work

**Comment by Dmitry-Ostashev** [comment]
@testcafe-build-bot retest

**Comment by Dmitry-Ostashev** [comment]
@testcafe-build-bot retest

File: test/functional/fixtures/ui/testcafe-fixtures/status-bar-test.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    await t
+        .expect(statusText.trim()).eql('Status prefix.')
+        .expect(statusDiv.innerText).eql('Status prefix. Waiting for assertion execution...')
+        .navigateTo('about:blank')
+        .expect(statusDiv.innerText).eql('Status prefix. Waiting for assertion execution...');
+});
+
+test('Hide elements when resizing the window', async t => {
+    const statusBarDiv = Selector(() => window['%testCafeDriverInstance%'].statusBar.statusBar);
+    const statusDiv    = Selector(() => window['%testCafeDriverInstance%'].statusBar.statusDiv);
```
Comment (helen-dikareva) [inline test/functional/fixtures/ui/testcafe-fixtures/status-bar-test.js]: (minor) let's exract statusBar selectors, place them in the begging of the test file and reuse them

**Pull request merged**